inspirehep/migrator/tasks.py

```diff
@@ -126,54 +106,71 @@ def migrate_record_from_legacy(recid):
-        wait_for_results(bool): flag indicating whether the task should wait
-            for the migration to finish (if True) or fire and forget the migration
-            tasks (if False).
+        disable_orcid_push (bool): flag indicating whether the orcid_push
+            sould be disabled (if True) or executed at the end of migrations (if False).
     """
-
     query = LegacyRecordsMirror.query.with_entities(LegacyRecordsMirror.recid)
+
     if also_migrate is None:
         query = query.filter(LegacyRecordsMirror.valid.is_(None))
     elif also_migrate == "broken":
         query = query.filter(LegacyRecordsMirror.valid.isnot(True))
     elif also_migrate != "all":
         raise ValueError('"also_migrate" should be either None, "all" or "broken"')
 
-    if wait_for_results:
-        # if the wait_for_results is true we enable returning results from the
-        # migrate_recids_from_mirror task so that we could use them to
-        # synchronize migrate task (which in that case waits for the
-        # migrate_recids_from_mirror tasks to complete before it finishes).
-        tasks = []
-        migrate_recids_from_mirror.ignore_result = False
-
-    chunked_recids = chunker(res.recid for res in query.yield_per(CHUNK_SIZE))
-    for i, chunk in enumerate(chunked_recids):
-        scheduled_records = i * CHUNK_SIZE + len(chunk)
-        print(f"Scheduled {scheduled_records} records for migration")
-        if wait_for_results:
-            tasks.append(migrate_recids_from_mirror.s(chunk))
-        else:
-            migrate_recids_from_mirror.delay(chunk)
-
-    if wait_for_results:
-        job = group(tasks)
-        result = job.apply_async()
-        result.join()
-        migrate_recids_from_mirror.ignore_result = True
-        print("All migration tasks have been completed.")
+    recids_chunked = chunker(res.recid for res in query.yield_per(CHUNK_SIZE))
+    tasks = migrate_recids_from_mirror(list(recids_chunked), 0, disable_orcid_push)
+    print("All migration  tasks has been scheduled.")
+    return tasks
+
+
+# Set time_limit for task to 24 hours
+@shared_task(ignore_results=True, queue="migrator")
+def migrate_recids_from_mirror(records_chunks, step_no=0, disable_orcid_push=True):
+    """Function to create group of tasks and adding them to chord."""
+    MIGRATION_STEPS = [
+        create_records_from_mirror_recids,
+        recalculate_citations,
+        index_record,
+    ]
+    if not disable_orcid_push:
+        MIGRATION_STEPS.append((run_orcid_push,))
+
+    if step_no >= len(MIGRATION_STEPS):
+        return
+
+    logger.info(f"Running step {step_no+1} from {len(MIGRATION_STEPS)}")
+
+    all_recs = 0
+    for row in records_chunks:
+        all_recs += len(row)
+    if all_recs == 0:
+        logger.warning("There are no records to migrate. Terminating.")
+        return None
+    logger.info(
+        f"Chunking... Creating {len(records_chunks)} tasks with {CHUNK_SIZE} records each"
+    )
```

Comment:
leftover? chunking isn't happening inside the task anymore

Reply:
:+1: